lib/stac-api/runtime/src/handler.py

```diff
@@ -6,8 +6,40 @@
 import os
 
 from mangum import Mangum
+from utils import get_secret_dict
+
+secret = get_secret_dict(secret_arn_env_var="PGSTAC_SECRET_ARN")
+
+os.environ.update(
+    {
+        "postgres_host_reader": secret["host"],
+        "postgres_host_writer": secret["host"],
+        "postgres_dbname": secret["dbname"],
+        "postgres_user": secret["username"],
+        "postgres_pass": secret["password"],
+        "postgres_port": str(secret["port"]),
+    }
+)
+
+from stac_fastapi.pgstac.app import app
+from stac_fastapi.pgstac.db import close_db_connection, connect_to_db
+
+
+@app.on_event("startup")
+async def startup_event():
+    """Connect to database on startup."""
+    print("Setting up DB connection...")
+    await connect_to_db(app)
```

Comment:
Sadly settings and `connect_to_db` are defined a bit differently in stac-fastapi-pgstac https://github.com/stac-utils/stac-fastapi-pgstac/blob/b1b393c6d283da0dd4e302fd53fb5f8740859d50/stac_fastapi/pgstac/db.py#L48-L62

it would be hard to refactor it in a way to avoid `os.environ` override :-(

Reply:
maybe we could add a `create_app` factory function to these applications which could be used to create an instance of `app` with some provided settings.

Reply:
We refactored the settings in stac-fastapi-pgstac with https://github.com/stac-utils/stac-fastapi-pgstac/pull/209